File: app/src/main/java/com/github/hitgif/powerscore/History.java
package com.github.hitgif.powerscore;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Created by 王安海 on 12/23/2015.
 */
public class History {
    public History(int _score, String _names, String _reason, Date _date, String _oper){
        score=_score;
        names =_names;
        reason =_reason;
        oper=_oper;
        //score_now=_score_now;
        date=_date;
        shortReason=reason.length() > 6 ? reason.substring(0, 6) + "…" : reason;
        shortNames=names.length() > 25 ? names.substring(0, 25) + "…" : names;
        scoreWithSign=(score > 0 ? "+" : "") + (score / 10.0);
        dateStr= new SimpleDateFormat("yyyy-MM-dd HH:mm:ss:SS", Locale.CHINA).format(date);
        shortDateStr=new SimpleDateFormat("MM-dd", Locale.CHINA).format(date);
    }
    public History(Date _date){
        score=0;
        names ="";
        reason ="";
        oper="";
        date=_date;
        shortReason="";
        shortNames="";
        scoreWithSign="";
        dateStr= new SimpleDateFormat("yyyy-MM-dd HH:mm:ss:SS", Locale.CHINA).format(date);
        shortDateStr="";
    }
    public final Date date;
    public final int score;
    public final String names;
    public final String reason;
    public final String oper;

    public final String shortReason;
    public final String shortNames;
    public final String scoreWithSign;

    private final String dateStr;
    private final String shortDateStr;

    public String getScore(){
        return (score>0?"+":"")+(score / 10.0);
    }
    public String getDate(boolean showAll){
        if(showAll) return dateStr;
        else return shortDateStr;
    }
}

File: PowerScore/History.swift
//
//  History.swift
//  PowerScore
//
//  Created by Harry on 16/7/7.
//  Copyright © 2016年 Harry Null. All rights reserved.
//

import Foundation

class History : NSObject{
    init(_score:Int, _names:String, _reason:String, _date:Date, _oper:String) {
        score = _score
        names = _names
        reason = _reason
        oper=_oper
        date=_date
        shortReason = reason.characters.count > 6 ? reason.substringWithRange(Range<String.Index>(start:reason.startIndex,end:reason.startIndex.advancedBy(6))) + "…" : reason
        shortNames = names.characters.count > 10 ? names.substringWithRange(Range<String.Index>(start:reason.startIndex,end:reason.startIndex.advancedBy(10))) + "…" : names
        scoreWithSign = (score > 0 ? "+" : "-") + String(Double(score) / 10.0)
        dateStr = date.stringWithFormat("yyyy-MM-dd HH:mm:ss:SS")
        shortDateStr = date.stringWithFormat("MM-dd")
    }
    func encodeWithCoder(aCoder: NSCoder){
        aCoder.encodeObject(self.date.timeInterval, forKey: "timeInterval")
        aCoder.encodeObject(self.score, forKey: "score")
        aCoder.encodeObject(self.names, forKey: "names")
        aCoder.encodeObject(self.reason, forKey: "reason")
        aCoder.encodeObject(self.oper, forKey: "oper")
    }
    
    required init(coder aDecoder: NSCoder) {
        let dateInterval = aDecoder.decodeObjectForKey("timeInterval") as! NSTimeInterval
        self.date=Date(dateInterval)
        self.score = aDecoder.decodeObjectForKey("score") as! Int
        self.names = aDecoder.decodeObjectForKey("names") as! String
        self.reason = aDecoder.decodeObjectForKey("reason") as! String
        self.oper = aDecoder.decodeObjectForKey("oper") as! String
        
        shortReason = reason.characters.count > 6 ? reason.substringWithRange(Range<String.Index>(start:reason.startIndex,end:reason.startIndex.advancedBy(6))) + "…" : reason
        shortNames = names.characters.count > 10 ? names.substringWithRange(Range<String.Index>(start:reason.startIndex,end:reason.startIndex.advancedBy(10))) + "…" : names
        scoreWithSign = (score > 0 ? "+" : "-") + String(Double(score) / 10.0)
        dateStr = date.stringWithFormat("yyyy-MM-dd HH:mm:ss:SS")
        shortDateStr = date.stringWithFormat("MM-dd")
    }
    let date:Date
    let score:Int
    let names:String
    let reason:String
    let oper:String
    
    let shortReason:String
    let shortNames:String
    let scoreWithSign:String
    
    let dateStr:String
    let shortDateStr:String
    
    func getRealScore()->String{
        return scoreWithSign
    }
    func getDateFast(showAll:Bool)->String{
        if(showAll){return dateStr}
        else {return shortDateStr}
    }
}
